internal/pkg/cli/deploy.go

```diff
@@ -243,31 +249,43 @@ func (o *deployOpts) Run() error {
 		return err
 	}
 
-	if err := o.maybeInitWkld(); err != nil {
-		return err
+	for _, workload := range o.workloadNames {
+		if err := o.maybeInitWkld(workload); err != nil {
+			return err
+		}
+		deployCmd, err := o.loadWkldCmd(workload)
+		if err != nil {
+			return err
+		}
+		if err := deployCmd.Ask(); err != nil {
+			return fmt.Errorf("ask %s deploy: %w", o.wlType, err)
+		}
+		if err := deployCmd.Validate(); err != nil {
+			return fmt.Errorf("validate %s deploy: %w", o.wlType, err)
+		}
+		if err := deployCmd.Execute(); err != nil {
+			return fmt.Errorf("execute %s deploy: %w", o.wlType, err)
+		}
+		if err := deployCmd.RecommendActions(); err != nil {
+			return err
+		}
 	}
 
-	if err := o.loadWkld(); err != nil {
-		return err
-	}
-	if err := o.deployWkld.Execute(); err != nil {
-		return fmt.Errorf("execute %s deploy: %w", o.wlType, err)
-	}
-	if err := o.deployWkld.RecommendActions(); err != nil {
-		return err
-	}
 	return nil
 }
 
 func (o *deployOpts) askName() error {
-	if o.name != "" {
+	if o.deployAllWorkloads {
+
+	}
+	if o.workloadNames != nil || len(o.workloadNames) != 0 {
```

Comment:
What's the UX going to be for multiple deployments? This appears to have the assumption that `askName()` will only be called when the user intends to only deploy one workflow, so I'm curious how the UX design takes this into account.

Reply:
`sel.Workload` will be replaced by `sel.Workloads` once #5308 is merged.